src/Parser/Serialization/FuncSerializer.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Text;
+
+namespace Parser
+{
+    public class FuncSerializer<T> : ISerialisation
+        where T : class //todo add also class for struct
```

Comment:
Can't we remove the `class` restriction (hence covering `struct`) by using a `is` + cast combo rather than `as` in the `SerializeObject` code?